modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -22,497 +22,683 @@ ADLAROMANCEStressUpdateBase::validParams()
+
+  // Prepare old values
+  _old_input_values[_cell_output_index] =
+      _cell_function ? _cell_function->value(_t, _q_point[_qp]) : _cell_dislocations_old[_qp];
+  _old_input_values[_wall_output_index] =
+      _wall_function ? _wall_function->value(_t, _q_point[_qp]) : _wall_dislocations_old[_qp];
+  _old_input_values[_strain_output_index] =
+      _creep_strain_old_forcing_function
+          ? _creep_strain_old_forcing_function->value(_t, _q_point[_qp])
+          : std::sqrt(_creep_strain_old[_qp].doubleContraction(_creep_strain_old[_qp]) / 1.5);
 
-  if (_verbose)
+  // Prepare input
+  _input_values[_cell_input_index] = _old_input_values[_cell_output_index];
+  _input_values[_wall_input_index] = _old_input_values[_wall_output_index];
+  _input_values[_stress_input_index] = effective_trial_stress * 1.0e-6;
+  _input_values[_old_strain_input_index] = _old_input_values[_strain_output_index];
+  _input_values[_temperature_input_index] = _temperature[_qp];
+  if (_environmental)
+    _input_values[_environmental_input_index] = (*_environmental)[_qp];
+
+  // Determine tile mixing weight and check to see if input is in range
+  std::fill(_non_stress_weights.begin(), _non_stress_weights.end(), 1.0);
+  for (unsigned int i = 0; i < _num_inputs; i++)
   {
-    Moose::out << "Verbose information from " << _name << ": \n";
-    Moose::out << "  dt: " << _dt << "\n";
-    Moose::out << "  old mobile disl: " << _mobile_old << "\n";
-    Moose::out << "  old immobile disl: " << _immobile_old << "\n";
-    Moose::out << "  initial stress (MPa): "
-               << MetaPhysicL::raw_value(effective_trial_stress) * 1.0e-6 << "\n";
-    Moose::out << "  temperature: " << MetaPhysicL::raw_value(_temperature[_qp]) << "\n";
-    Moose::out << "  environmental factor: " << MetaPhysicL::raw_value(_environmental[_qp]) << "\n";
-    Moose::out << "  calculated scalar strain value: " << MetaPhysicL::raw_value(scalar) << "\n";
-    Moose::out << "  trial stress into rom (MPa): " << MetaPhysicL::raw_value(trial_stress_mpa)
-               << "\n";
-    Moose::out << "  old effective strain: " << effective_strain_old << "\n";
-    Moose::out << " ROM outputs \n";
-    Moose::out << "  effective incremental strain from rom: "
-               << MetaPhysicL::raw_value(rom_effective_strain) << "\n";
-    Moose::out << "  new effective strain: "
-               << MetaPhysicL::raw_value(effective_strain_old + rom_effective_strain) << "\n";
-    Moose::out << "  new mobile dislocations: "
-               << _mobile_old + MetaPhysicL::raw_value(_mobile_dislocation_increment) << "\n";
-    Moose::out << "  new immobile dislocations: "
-               << _immobile_old + MetaPhysicL::raw_value(_immobile_dislocation_increment) << "\n"
-               << std::endl;
+    if (i != _stress_input_index)
+    {
+      computeTileWeight(_non_stress_weights, _input_values[i], i);
+      checkInputWindow(_input_values[i], _window_failure[i], _global_limits[i]);
```

Comment:
I haven't added a bunch of print statements to this yet, but it seemed like this check wasn't being enforced for the dislocations in my simulation. Not worth holding up this merge request but something that I'd like to look into on the next iteration